docs/reference/contributing/target/bootstrap.md

```diff
@@ -282,15 +287,15 @@ SECTIONS
 }
 ```
 
-#### Create and update files
+##### Other required files
 
-- Extend CMSIS-CORE by adding the file `mbed-os\targets\TARGET_VENDOR\TARGET_MCUNAME\cmsis.h`. This header file includes device-specific headers that include CMSIS-CORE.
-- Add a relocation function in `mbed-os\targets\TARGET_VENDOR\TARGET_MCUNAME\cmsis_nvic.h`. This contains the define `NVIC_NUM_VECTORS`, which is the number of vectors the devices has, and `NVIC_RAM_VECTOR_ADDRESS`, which is the address of the RAM vector table.
+- Make sure your CMSIS-Core implementation contains the <a href="http://www.keil.com/pack/doc/CMSIS/Core/html/device_h_pg.html" target="_blank">`device.h` header</a>.
+- Extend CMSIS-Core by adding the file `mbed-os\targets\TARGET_VENDOR\TARGET_MCUNAME\cmsis.h`. This header file includes device-specific headers that include CMSIS-Core. It must also include `cmsic_nvic.h`.
+- Add the `mbed-os\targets\TARGET_VENDOR\TARGET_MCUNAME\cmsis_nvic.h` header file. This contains the define `NVIC_NUM_VECTORS`, which is the number of vectors the devices has, and `NVIC_RAM_VECTOR_ADDRESS`, which is the address of the RAM vector table. Mbed OS relocates the vectors from the initial location in ROM to the provided address in RAM and updates the `VTOR` register. NOTE: For devices without the `VTOR` register, you need to make sure the vectors are in the read-write memory before execution reaches the `main` function. In this case, you may also need to provide visualization of NVIC access functions. For details, please see the <a href="https://www.keil.com/pack/doc/CMSIS/Core/html/group__NVIC__gr.html" target="_blank">CMSIS NVIC documentation</a>.
 - Define the initial stack pointer, `INITIAL_SP`, in `mbed_rtx.h`. This file is typically in `mbed-os\targets\TARGET_VENDOR\mbed_rtx.h`.
-- Add an entry for your device in `mbed-os\targets\targets.json`
 
-#### Verification and next steps
+### Entry points
 
-Now verify that your target compiles by using `mbed compile -m MCU_NAME -t <toolchain>`.
+Except the reset vector, which is the standard entry point for Cortex-M cores, Mbed OS provides `mbed_sdk_init`, which the target can overload to perform higher level initialization. This function will be called later in the bootstrap process, after the basic initialization is done but before RTOS starts and before the `main` function is called.
 
-The next step to port a target is to enable the test harness dependencies. To run the test suite, your target must support GPIO, microsecond ticker and serial.
+Mbed OS provides another entry point that will be executed before `main` called `mbed_main`. This function is reserved for application use, and the target code should not define it.
```

Comment:
Query: Who or what executes the entry point before `main`? Also, who or what reserves the function for application use?

Reply:
main is called by Mbed OS internals. As to who reserves the function it's 'us' as in we provide two functions `mbed_main` and `mbed_sdk_init` and we want one of them be available for platform code and second for the application code.